rust/lance/src/dataset/scanner.rs

```diff
@@ -283,11 +282,16 @@ impl Scanner {
     /// Once the filter is applied, Lance will create an optimized I/O plan for filtering.
     ///
     pub fn filter(&mut self, filter: &str) -> Result<&mut Self> {
-        parse_sql_filter(filter)?;
-        self.filter = Some(filter.to_string());
+        let planner = Planner::new(self.schema()?);
```

Comment:
Is `self.schema()` the output schema of the scanner or the schema of the dataset?  Will this work if the user does something like...

```
scan.project(&["a"]).filter("b < 10");
```

Reply:
Schema of the dataset. It's fine as long as this is left to a logical expression. Once we translate to a physical expression, then it gets converted to ordinal column references, which would break here. But that's why we convert to physical expressions later.